osquery/tables/networking/aws_metadata.h

```diff
@@ -0,0 +1,113 @@
+/*
+ *  Copyright (c) 2016-present, Jean-Francois Dive <jfdive@gmail.com>
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#pragma once
+
+#include <string>
+
+#include "include/osquery/tables.h"
+
+namespace osquery {
+namespace tables {
+
+/**
+ * @brief SuperClass for AWS metadata accessors
+ */
+class AwsData {
```

Comment:
A neat interface we use often is `boost::noncopyable` found in `<boost/noncopyable.hpp>`. It's an easy thing to inherit and clearly explains how you intend the data structure to be used. I usually recommend this regardless of the danger of copying pointers.